### PR TITLE
Form validates dates and property

### PR DESCRIPTION
**Pull request by yunnie**
### What issue is this solving?
Please connect this Pull Request to the issue it is resolving by using the controls on the right side of this PR.
#579 

### Any helpful knowledge/context for the reviewer?
Is a `npm install` necessary? No
Any special requirements to test? No

### Please make sure you've attempted to meet the following coding standards
- [x ] Code builds successfully
- [x ] Code has been tested and does not produce errors
- [x ] Code is readable and formatted
- [x ] There isn't any unnecessary commented-out code
- [x ] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

If you're having trouble meeting this criteria, feel free to reach out on Slack for help!

### CODE REVIEW

**Comment by yunnie** [comment]
Changed the validation so that all validation is done via Formik. If a property is selected the lease start and end dates will be required. If lease start and end dates are provided, then a property is required.

Removed some state hooks that are no longer necessary because all validation is done via Formik.

Also used the Formik `ErrorMessage` component to display any validation errors.

**Comment by aedwardg** [comment]
@yunnie, 

I really like what you did with this and I think it's a big improvement already.
However, it looks like it still doesn't cover the first problem I mentioned in the issue:
> incomplete lease info (filled in unit num and/or occupants, but missing dates), and no property info

If the user puts in a unit number/occupants but leaves the lease dates and property blank, the form still submits, creating a tenant and failing silently on the lease. 

I guess the question is: is this what we want to happen? Or do we want to prevent the form from submitting? Or do we want to warn the user that the lease won't be created if they continue?

**Comment by yunnie** [comment]
It does make sense to assume that if the user enters a Unit Number of a Number of Occupants that they intend to enter a lease, in which case a property and lease start and end dates are also required. If the entry of unit number and/or number of occupants was accidental, then validation can alert the user to the accidental entries (via error messages stating that property and lease dates are also required). I'll make the change to the validation.

File: src/views/AddTenant/addTenant.jsx [hunk 1]
```diff
@@ -143,16 +138,35 @@ export const AddTenant = () => {
...
+        errors.lease = "Lease dates required when a property is selected"
+      }
+    }
     if (dateTimeStart !== dateTimeEnd) {
-      setPropertyErrorText(dateTimeStart !== dateTimeEnd && propertySelection.length
-        ? "" 
-        : "*Property is a required field"
-      );
+      if (!propertySelection.length) {
+        errors.propertySelection = "Property is required when lease dates are slected"
```
Comment (aedwardg) [inline src/views/AddTenant/addTenant.jsx]:
```suggestion
        errors.propertySelection = "Property is required when lease dates are selected"
```

**Review by aedwardg — COMMENTED** [review]
@yunnie This is great. It's a very graceful way to handle the validation and I feel pretty confident that after this change even the most clueless of users will be able to properly fill out this form.

I did find one tiny typo, but other than that, this looks good to go

**Review by aedwardg — APPROVED** [review]
Looks good to me!

**Review by Chris-Boe — APPROVED** [review]
Nice work!
I noticed a couple things were broken that were not part of this PR so I'll create separate issues for them

**Pull request merged**